File: src/main/java/de/maximilianwollnik/homepage/model/Biography.java
package de.maximilianwollnik.homepage.model;

import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.Date;

/**
 * The type Biography.
 */
@Data
@EqualsAndHashCode(callSuper=false)
public class Biography extends Education {
    private BiographyState biographyState;

    /**
     * Instantiates a new Biography.
     *
     * @param start   the start
     * @param end     the end
     * @param element the element
     * @param biographyState the biographyState
     */
    public Biography(Date start, Date end, String element, BiographyState biographyState) {
        super(start, end, element);
        this.biographyState = biographyState;
    }
}

File: src/main/frontend/libs/data/src/lib/biography.ts
import { Item } from '..';

export enum BiographyState {
  JOB_FINISHED = 'JOB_FINISHED',
  JOB_CURRENT = 'JOB_CURRENT',
  PROJECT_FINISHED = 'PROJECT_FINISHED',
  PROJECT_CURRENT = 'PROJECT_CURRENT',
}

export interface Biography extends Item {
  start: Date;
  end: Date;
  biographyState: BiographyState;
}
